hack/generator/pkg/jsonast/schema_abstraction.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package jsonast
+
+import (
+	"net/url"
+)
+
+// Schema abstracts over the exact implementation of
+// JSON schema we are using as we need to be able to
+// process inputs provided by both 'gojsonschema' and
+// 'go-openapi'. It is possible we could switch to
+// 'go-openapi' for everything because it could handle
+// both JSON Schema and Swagger; but this is not yet done.
+type Schema interface {
```

Comment:
It needs to be exported to be able to invoke `scanner.GenerateDefinitions` with a `Schema`. You make `Schema` instances with the two factory methods `MakeGoJSONSchema` and `MakeOpenAPISchema`.

Reply:
Reactivated this because it feels weird to have an exported interface but no expected methods (I had the same double-take as @babbageclunk)

Is the _intent_ to hide the shape of this thing from callers? (i.e. they can make on using `MakeGoJSONSchema` or `MakeOpenAPISchema` but they aren't really supposed to be able introspect the type they get back?)

Not saying this needs to change, just want to double check that we're in a situation where we are hiding the internals because its "the right thing to do" and not just because this so happens to have been defined in the same module where it's used.

Reply:
yes, that is the intent, the internals are not useful for the caller since they’re just for `jsonast` to use (and exposing them means documenting them all as well!)

Reply:
As long as that's the intent I think it's fine.